### PR TITLE
Add script that hides persists preview bar minimize

### PR DESCRIPTION
**Pull request by t-kelly**
To 🎩 :
1. Switch to this branch on your computer
2. Change the reference of Slate Tools reference in a Slate theme `package.json` so that it points to your local Slate Tools repo, e.g. `"@shopify/slate-tools": "file:../slate-tools"`
3. Remove your theme's `node_modules` and reinstall: `rm -rf node_modules && npm install`
4. Run `slate watch` and test the minimize button. 

### What are you trying to accomplish with this PR?
Fixes https://github.com/Shopify/slate/issues/141

### Checklist
For maintainers:
- [x] I have :tophat:'d these changes.
- [ ] I have bumped the `package.json` version in a separate PR, if applicable.

### CODE REVIEW

File: src/tasks/deploy-sync.js [hunk 1]
```diff
@@ -51,6 +54,15 @@ gulp.task('deploy:sync-init', () => {
         next();
       },
     },
+    snippetOptions: {
+      // Provide a custom Regex for inserting the snippet.
+      rule: {
+        match: /<\/body>/i,
+        fn: (snippet, match) => {
+          return `<script defer="defer">${devScript}</script>${snippet}${match}`;
+        },
```
Comment (chrisberthe) [inline src/tasks/deploy-sync.js]: This is cool 👍.

File: src/tasks/includes/dev-script.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+      hidePreviewBar();
+    }
+  })
+
+  function onButtonClick(event) {
+    var element = event.target;
+
+    if (element.className.indexOf('shopify-preview-bar__minimize') === -1) { return; }
+
+    window.sessionStorage.setItem('hide-preview', 'true');
```
Comment (chrisberthe) [inline src/tasks/includes/dev-script.js]: Suggestion: `preview-bar-hidden`?

File: src/tasks/includes/dev-script.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    hidePreviewBar();
+    document.removeEventListener('click', onButtonClick);
+  }
+
+  function hidePreviewBar() {
+    addCSS('.shopify-preview-bar { display:none; }');
+  }
+
+  function addCSS(css){
```
Comment (chrisberthe) [inline src/tasks/includes/dev-script.js]: Suggestion: `injectStyles(css)`?

File: src/tasks/includes/dev-script.js [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    if (style.styleSheet) {   // IE
+      style.styleSheet.cssText = css;
+    } else {                // Everything else
+      style.appendChild(document.createTextNode(css));
+    }
+
+    head.appendChild(style);
+  }
+
+  function isLocalStorageSupported () {
```
Comment (chrisberthe) [inline src/tasks/includes/dev-script.js]: `isSessionStorageSupported()`

File: src/tasks/includes/dev-script.js [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    var mod = 'modernizr';
+    try {
+      localStorage.setItem(mod, mod);
+      localStorage.removeItem(mod);
```
Comment (chrisberthe) [inline src/tasks/includes/dev-script.js]:
```js
sessionStorage.setItem(mod, mod);
sessionStorage.removeItem(mod);
```

File: src/tasks/includes/dev-script.js [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+ * Development Script
+ *
+ * This script is only included on the page when Browsersync is being used. It's
+ * a great place to put any customizations that you only want to occur while
+ * developing your theme.
+ ******************************************************************************/
+
+
+/**
+ * Persistent Preview Bar Minimize
```
Comment (chrisberthe) [inline src/tasks/includes/dev-script.js]: nit: sentence case (same for `Development script` above)

File: src/tasks/includes/dev-script.js [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+ */
+
+(function() {
+  if (!isSessionStorageSupported()) { return; }
+
+  window.addEventListener('DOMContentLoaded', function() {
+    document.addEventListener('click', onButtonClick);
+
+    if (window.sessionStorage.getItem('hide-preview')) {
+      hidePreviewBar();
```
Comment (huguestennier) [inline src/tasks/includes/dev-script.js]: Needs to be `preview-bar-hidden`

Comment (chrisberthe) [inline src/tasks/includes/dev-script.js]: True, good catch

**Comment by daviddarnes** [comment]
Sorry to bother, but is this change working as expected? I'm running `slate watch` with slate 0.11.0 and slate-tools 0.3.3 and the preview bar is removed from the dom rather than hidden with injected css. Navigating anywhere in the theme for me means the preview pane still re-appears?

**Comment by t-kelly** [comment]
Hey @daviddarnes thanks for checking this out! Looks like you're right and the `0.11.0` branch this merged into never got merged into master.

I'm opening up another PR to have these changes merged into the next version of Slate. Sorry for the hiccup!

**Comment by daviddarnes** [comment]
@t-kelly awesome! No need to be sorry, thank you for contributing / creating this tool!

**Pull request merged**